src/plugins/persistence/couch/CouchObjectProvider.js

```diff
@@ -40,6 +41,62 @@ export default class CouchObjectProvider {
         this.batchIds = [];
     }
 
+    /**
+     * @private
+     */
+    startSharedWorker() {
+        let provider = this;
+        let sharedWorker;
+
+        sharedWorker = new SharedWorker('data:application/javascript;base64,' + btoa(CouchChangesFeed));
```

Comment:
Base64 encoded URL can be used for SharedWorkers to work without having to change markup in index.html
However, There's a limit to the length of the URL :/ 

I'm looking at other options.

Reply:
This was resolved by adding an additional asset for the shared worker code and using the asset path to get the predictable URL for SharedWorker. Thanks @akhenry for the suggestion.